lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -189,18 +190,18 @@ class ToLayoutOpConversionPattern
     // device->host path
     //
     if (bufferType == ttnn::BufferType::SystemMemory) {
-      ttnn::FromDeviceOp fromDeviceOp = rewriter.create<ttnn::FromDeviceOp>(
+      // Change to RowMajor
+      //
+      ttnn::ToLayoutOp toLayoutOp = rewriter.create<ttnn::ToLayoutOp>(
           op.getLoc(), this->getTypeConverter()->convertType(op.getType()),
-          op->getOperand(0));
+          op.getOperand(0), device,
+          ttnn::LayoutAttr::get(op->getContext(), ttnn::Layout::RowMajor));
```

Comment:
If the operand is already row major maybe we don't need to emit this layout op that would otherwise be a nop?

Reply:
@svuckovicTT is on PTO right now, and assigned me to check in this change after it's approved. Is it okay to leave it for him to fix toLayout no-op when he gets back? I can create a follow up issue.

Reply:
Chiming in from the airport - cheers to delayed flights! :)

Yeah, we could avoid it, I'll create a follow-up task (can't VPN from Hungary, so can't test the runtime...). However, I think it's okay to check this in as-is for now, since it'll end up being a nop - the method checks for this and returns if the layout is already set to the desired one.

Reply:
https://github.com/tenstorrent/tt-mlir/issues/773